sklearn/impute.py

```diff
@@ -825,8 +846,13 @@ def fit_transform(self, X, y=None):
                                      check_random_state(self.random_state))
 
         if self.predictor is None:
-            from .linear_model import BayesianRidge
-            self._predictor = BayesianRidge()
+            if self.sample_posterior:
+                from .linear_model import BayesianRidge
+                self._predictor = BayesianRidge()
+            else:
+                from .linear_model import RidgeCV
+                # including a very small alpha to approximate OLS
+                self._predictor = RidgeCV(alphas=np.array([1e-5, 0.1,  1, 10]))
         else:
             self._predictor = clone(self.predictor)
```

Comment:
Actually we should set the state here also if `_predictor` has a `random_state`.

Reply:
Right, good point.

Reply:
Actually, shouldn't that be done on line 857 instead? I.e. just once?
```
        if self.predictor is None:
            if self.sample_posterior:
                from .linear_model import BayesianRidge
                self._predictor = BayesianRidge()
            else:
                from .linear_model import RidgeCV
                # including a very small alpha to approximate OLS
                self._predictor = RidgeCV(alphas=np.array([1e-5, 0.1,  1, 10]))
        else:
            self._predictor = clone(self.predictor)
        # SET RANDOM STATE HERE
```

Reply:
It is what I meant.